### PR TITLE
solved task

### PR DESCRIPTION
**Pull request by slaybrute**

### CODE REVIEW

File: src/main/java/core/basesyntax/Figure.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    public void printInfoAboutFigures() {
+        System.out.println("Figure: " + name + ", " + "color: " + color
+                + ", " + "area: " + area + " square units, ");
+    }
+
+    public void setArea(double area) {
+        this.area = area;
+    }
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/Figure.java:21]: This methods should not be implemented in this abstract class. Implement it in your figures classes
```suggestion
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    public static String getRandomColor() {
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]:
```suggestion
    public String getRandomColor() {
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+        Random random = new Random();
+        String[] colors = new String[]{"red", "blue", "green", "white", "black", "yellow", "brown"};
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/ColorSupplier.java:8]: Create enum for Color
```suggestion
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    public static String getRandomColor() {
+        Random random = new Random();
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: Create Random as a class field

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier extends ColorSupplier {
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/FigureSupplier.java:5]: FigureSupplier is not a ColorSupplier
```suggestion
public class FigureSupplier {
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    public static Figure getRandomFigure() {
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/FigureSupplier.java:6]:
```suggestion
    public Figure getRandomFigure() {
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        Random random = new Random();
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/FigureSupplier.java:7]: Create Random as a class field

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+        Figure[] figures = new Figure[]{new Circle("circle", getRandomColor(), random.nextInt(26)),
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/FigureSupplier.java:8]: It's better to use "switch" here.

switch (index) {
            case 0:
                return getRandomCircle();
            case 1:
                return getRandomIsoscelesTrapezoid();
            case 2:
                return getRandomRectangle();
            case 3:
                return getRandomRightTriangle();
            case 4:
            default:
                return getRandomSquare();
        }

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+                        random.nextInt(26), random.nextInt(26)),
+                new IsoscelesTrapezoid("isosceles trapezoid", getRandomColor(),
+                        random.nextInt(26), random.nextInt(26),
+                        random.nextInt(26), random.nextInt(26)),
+                new Square("square", getRandomColor(),
+                        random.nextInt(25))};
+        return figures[random.nextInt(figures.length)];
+    }
+
+    public static Figure getDefaultFigure() {
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/FigureSupplier.java:21]:
```suggestion
    public Figure getDefaultFigure() {
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+        return new Circle("circle", "white", 10);
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/FigureSupplier.java:22]: Create constants
```suggestion
        return new Circle(DEFAULT_COLOR, DEFAULT_RADIUS);
```

File: src/main/java/core/basesyntax/Circle.java [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+
+    public Circle(String name, String color, int radius) {
+        super(name, color);
+        this.radius = radius;
+        super.setArea(Math.PI * Math.pow(radius, 2));
+    }
+
+    @Override
+    public void printInfoAboutFigures() {
+        super.printInfoAboutFigures();
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/Circle.java:14]: Don't implement this method in abstract Figure class
```suggestion
```

File: src/main/java/core/basesyntax/Circle.java [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+        System.out.println("radius " + radius);
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/Circle.java:15]: Put more information about Circle here

File: src/main/java/core/basesyntax/Rectangle.java [hunk 13]
```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/Rectangle.java:3]: Fix this class. Use class Circle as an example

File: src/main/java/core/basesyntax/RightTriangle.java [hunk 14]
```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/RightTriangle.java:3]: Fix this class. Use class Circle as an example

File: src/main/java/core/basesyntax/Square.java [hunk 15]
```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Square extends Figure {
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/Square.java:3]: Fix this class. Use class Circle as an example

File: src/main/java/core/basesyntax/IsoscelesTrapezoid.java [hunk 16]
```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/IsoscelesTrapezoid.java:3]: Fix this class. Use class Circle as an example

**Review by aleksandr-bevza — CHANGES_REQUESTED** [review]
Request changes

**Comment by slaybrute** [comment]
I've added changes and pushed them to this repository, tell me please do you see them ?

**Pull request closed without merging**